spirv_glsl.cpp

```diff
@@ -2409,7 +2409,26 @@ void CompilerGLSL::emit_quaternary_func_op(uint32_t result_type, uint32_t result
 	inherit_expression_dependencies(result_id, op3);
 }
 
-string CompilerGLSL::legacy_tex_op(const std::string &op, const SPIRType &imgtype)
+// EXT_shader_texture_lod only concerns fragment shaders so lod tex functions
+// are not allowed in ES 2 vertex shaders. But SPIR-V only supports lod tex
+// functions in vertex shaders so we revert those back to plain calls when
+// the lod is a constant value of zero.
+bool CompilerGLSL::check_explicit_lod_allowed(uint32_t lod)
+{
+	auto &execution = get_entry_point();
+	bool allowed = !is_legacy_es() || execution.model == ExecutionModelFragment;
+	if (!allowed)
+	{
+		auto *lod_constant = maybe_get<SPIRConstant>(lod);
+		if (!lod_constant || lod_constant->scalar_f32() != 0.0f)
```

Comment:
Because it can throw if lod is 0 and not a constant? I think that's ok because in that case the original code did use an explicit lod value (which turned out to be 0).